service/dap/server.go

```diff
@@ -742,7 +749,55 @@ func (s *Server) onLaunchRequest(request *dap.LaunchRequest) {
 		return
 	}
 
-	if mode == "debug" || mode == "test" {
+	// On replay mode, validate its required parameters to define the debugger
+	// Backend and CoreFile properties
+	if mode == "replay" {
+		// Check against replay parameters and decide the backend based on which one was received
+		coreDumpPath, _ := request.Arguments["coreDumpPath"].(string)
+		traceDirectory, _ := request.Arguments["traceDirectory"].(string)
+
+		if (coreDumpPath != "" && traceDirectory != "") || (coreDumpPath == "" && traceDirectory == "") {
+			s.sendErrorResponse(request.Request, FailedToLaunch, "Failed to launch", "The 'replay' mode must have either a coreDumpPath or a traceDirectory, but not both." )
+			return
+		}
+
+		if coreDumpPath != "" {
+			s.config.Debugger.CoreFile = coreDumpPath
+			s.config.Debugger.Backend = "core"
```

Comment:
Should something like this suffice? maybe the s.config.Debugger.Backend != "" check is redundant at this point since I should always expect "default" but I'm not sure about the later

```go
		var isBackendOverwritten bool
		if s.config.Debugger.Backend != "" || s.config.Debugger.Backend != "default" {
			isBackendOverwritten = true
		}
		if coreDumpFilePath != "" {
			s.config.Debugger.CoreFile = coreDumpFilePath
			s.config.Debugger.Backend = "core"
		} else {
			s.config.Debugger.CoreFile = traceDirectory
			s.config.Debugger.Backend = "rr"
		}
		if isBackendOverwritten {
			s.log.Debugf("overriding debugger backend to %s", s.config.Debugger.Backend)
		}
```

Reply:
I was thinking something simple like in case of switching goroutines, where you get the notification even if nothing  changed
```
(dlv) goroutine 1
Switched from 1 to 1 (thread 3416787)
```
If you really want to, you can check that new value if not the same as old value. But I don't see why you need to single out "" and "default". What if the value is something else? Why wouldn't you want logging then?

On a related note, I took another look at command.go and debugger.go. I think with coredumps the backend is basically ignored (if CoreFile is set, then backend is checked for being "rr" or anything else), so do you even need to set it at all? I am not sure what happens with "dlv core --backend=rr" though. Normally we would cover various valid/invalid flag combinations in server_test, so this would be clear.